Libraries/LibJS/AST.h

```diff
@@ -670,9 +670,30 @@ class VariableDeclaration : public Declaration {
     NonnullRefPtrVector<VariableDeclarator> m_declarations;
 };
 
+class ObjectProperty final : public ASTNode {
+public:
+    ObjectProperty(NonnullRefPtr<Expression> key, NonnullRefPtr<Expression> value)
+        : m_key(key)
+        , m_value(value)
+    {
+    }
+
+    const NonnullRefPtr<Expression> key() const { return m_key; }
+    const NonnullRefPtr<Expression> value() const { return m_value; }
```

Comment:
No need to return a `NonnullRefPtr` copy here as we are not giving up ownership of these objects.
```suggestion
    const Expression& key() const { return m_key; }
    const Expression& value() const { return m_value; }
```